lms/views/admin/application_instance/update.py

```diff
@@ -53,26 +51,23 @@ def update_instance(self):
             developer_secret=self.request.params.get("developer_secret", "").strip(),
         )
 
-        for (
-            setting,
-            sub_setting,
-        ), setting_type in APPLICATION_INSTANCE_SETTINGS.items():
-            value = self.request.params.get(f"{setting}.{sub_setting}")
+        for field in ApplicationSettings.fields:
+            value = self.request.params.get(field.compound_key)
             value = value.strip() if value else None
 
-            if setting_type is asbool:
+            if field.format is asbool:
```

Comment:
If we went with something like having `field.type` and `field.formatted` we could move these too `field.store(ai.settings, value)` or similar and use the type info there to store the setting accordingly.

Needing aes_service would make it a bit more awkward to probably something that could be left for another PR or refactor in the future.

Reply:
I think revisiting this again if we ever have the stomach to think about pushing this into the settings object would be a good time?